chatgpt-shell-ollama.el

```diff
@@ -84,6 +86,65 @@ VALIDATE-COMMAND handler."
          :token-width 4
          :context-window 32768)))
 
+(defun chatgpt-shell-ollama--fetch-model-versions ()
+  (mapcar (lambda (model)
+            (string-remove-suffix ":latest" (map-elt model 'name)))
+          (map-elt (shell-maker--json-parse-string
+                    (map-elt (shell-maker-make-http-request
+                              :async nil
+                              :url (concat chatgpt-shell-ollama-api-url-base "/api/tags"))
+                             :output))
+                   'models)))
+
+(defun chatgpt-shell-ollama--parse-token-width (quantization)
+  (string-match "^[FQ]\\([1-9][0-9]*\\)" quantization)
```

Comment:
We should prolly return nil if failed, so it can be handled later on.

```
(when-let ((matched (string-match "^[FQ]\\([1-9][0-9]*\\)" quantization)))
    (string-to-number (match-string 1 quantization)))
```